### ISSUE TITLE
Remove card mod and implement Advanced Report Hack

### ISSUE THREAD

**Issue by vendethiel** [IntegraMOD/IntegraMOD151#62]
I won't beat around the bush. The card mod is too dated, not intuitive, and not very helpful.

S2B's Advanced Report Hack mod looks very potent, and I'd like to integrate it, so that IntegraMOD can have real reporting tools.
What do you think?

**Comment by vendethiel** [comment]
Obviously I have to deal with PCP, as this modifies quite a bit of usercp_viewprofile..

**Comment by vendethiel** [comment]
QBar is too simple for our use case. I need to add more features to it (like being able to specify a PHP function). I want to know, before I do that, because I'm wary of security:

Is PCP available to every single admin on the board? That seems so dangerous... It seems like it should be founder-restricted.

**Comment by MWE001** [comment]
Items in the ACP is only available according to what permissions you give a User as a Jr Admin on the forum. It is up the Admin to be smart about his or her job and not dish out permissions like Monty Hall on the old Let's Make a Deal (did I just show some age on me?). 

And Qbar being "Simple" is for a reason. It was only ever meant to be used for creating menus as far as I can tell. Mountains can be made out of any mole hill if we try hard enough :-) but in the end, it was simply a menu feature and one that worked well at that. It did it's job for the intended purpose at the time. Now it is dated really bad. No sub menus? Oops. Got to have sub Menus. It is pleasing, keeps a board under control (no menu blocks the length of the entire website). 

I am curious to see what an Advanced Qbar mod could do though.

**Comment by MWE001** [comment]
And I figured when you mentioned removing Card Mod that you was in for work. Card Mod is integrated so far I often compared it to CH or PCP. :-) I have never saw the other Mod you mentioned.

**Comment by vendethiel** [comment]
The « advanced qbar » I made only has one extra feature: PHP can be used to make it dynamic.
Default qbar can only be dynamic with regards to PM, admin, and login/logout.

And while card mod is heavily integrated, it is 1/20 of the size of PCP, which means it’s (around) 20 times simpler to remove it


The reports mod is the screenshot I posted earlier. This is what the interface looks like when someone reports something.

**Comment by vendethiel** [comment]
Who am I kidding. It’s more like 1/50 of pcp...

**Comment by MWE001** [comment]
Exactly. That it is. I always hated doing a manual install of it though. it sucked @ss. Not quite as bad as PC or CH but it made me throw a choice word out more than a few times.

**Comment by vendethiel** [comment]
Well this time I'm uninstalling. It's harder though because I don't know how many things were changed in PCP etc.. to make it work with IM.

Did you look at my screen capture I posted earlier? What do you think about it?

**Comment by vendethiel** [comment]
Actually since Advanced Report Hack has no way to keep track of how many times a user was warned.. I'll do this instead: Keep Yellow Card mod, and integrate it with the Report Hack Mod.

I'll remove the "report" feature of yellow card however.

**Comment by vendethiel** [comment]
Perfect. I will be integrating both with the color username mod, as well as the report mod.

**Comment by MWE001** [comment]
That sounds good. It looks good thus far.

**Comment by vendethiel** [comment]
An issue I have with the Yellow Card mod is that you have a number of cards, but you don't know who, when, why, or where. I plan to add them, probably in a "card_history" table.

My goal is to have the Report Page from Advanced Report Hack show "this user already has [X] yellow cards <click here to see them>"

**Comment by vendethiel** [comment]
OK, first commit wiht 288e6000e2b353aaea4bd83a031ab77279fc93b9

**Comment by MWE001** [comment]
That would actually be pretty sweet. Would it include for what reason? Or just how many they have? I ask that because I have been an admin on many forums before and some mods/admins were power hungry and took it out on members that they "disliked" and cards were handed out that probably should not have been. If it had a mandatory uneditable reason of some type (I'm shooting at the moon here) that an admin could see ..... Now I'm grasping at straws and should be quiet. :-D

**Comment by vendethiel** [comment]
This is exactly the plan actually. I feel like phpBB in general really dropped the ball on moderation features. It's gotten better but we need to catch up by ourselves.

Any time you want to inflict a card, you need to give a reason.

When you check an user profile, you see the cards & why. Along with a link to all reports by user & about users.

**Comment by vendethiel** [comment]
Maybe even for a moderator, it could list all the inflicted cards.

**Comment by vendethiel** [comment]
Annoying part about Advanced Report Hack...
It doesn’t have a « who’s this report about ».
This is not always possible but it should be in most cases (except for general reports).

I’ll have to implement that

**Comment by vendethiel** [comment]
OK, well.
![](http://puu.sh/CEim5/3215b7736a.png)

I just implemented that!
Now the code remembers who was the report about, the "reportee". It also remembers the username in case the user is ever deleted. And you can even filter reports to find all about that specific user!

**Comment by vendethiel** [comment]
This only needs a fix to work with PCP and is done
